oracle/oracle.go

```diff
@@ -214,14 +215,21 @@ func (o *OracleImpl) fetchPrices(provider providertypes.Provider[oracletypes.Cur
 		o.logger.Debug(
 			"adding price",
 			zap.String("provider", provider.Name()),
+			zap.String("data handler type", string(provider.Type())),
 			zap.String("pair", pair.String()),
 			zap.String("price", result.Value.String()),
 			zap.Duration("diff", diff),
 		)
 		timeFilteredPrices[pair] = result.Value
+
+		// update price metric
+		o.metrics.UpdatePrice(provider.Name(), string(provider.Type()), pair.String(), float64(result.Value.Int64()))
```

Comment:
hmm, i wonder if it makes more sense to just log this price no matter what like on line 202 right before we filter, and then use the other metric to record whether this pair was used in aggregation. I'd prefer to log the prices as much as possible.